MapboxNavigation/RouteVoiceController.swift

```diff
@@ -121,19 +121,38 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate, AVAudioP
     
     deinit {
         suspendNotifications()
-        speechSynth.stopSpeaking(at: .word)
+        speechSynth.stopSpeaking(at: .immediate)
     }
     
     func resumeNotifications() {
         NotificationCenter.default.addObserver(self, selector: #selector(didPassSpokenInstructionPoint(notification:)), name: RouteControllerDidPassSpokenInstructionPoint, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(pauseSpeechAndPlayReroutingDing(notification:)), name: RouteControllerWillReroute, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(didReroute(notification:)), name: RouteControllerDidReroute, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(settingsDidChange(_:)), name: NavigationSettingsDidChange, object: nil)
     }
     
     func suspendNotifications() {
         NotificationCenter.default.removeObserver(self, name: RouteControllerDidPassSpokenInstructionPoint, object: nil)
         NotificationCenter.default.removeObserver(self, name: RouteControllerWillReroute, object: nil)
         NotificationCenter.default.removeObserver(self, name: RouteControllerDidReroute, object: nil)
+        NotificationCenter.default.removeObserver(self, name: NavigationSettingsDidChange, object: nil)
+    }
+    
+    @objc func settingsDidChange(_ notification: Notification) {
+        let muteKey = #keyPath(NavigationSettings.voiceMuted)
+        let volumeKey = #keyPath(NavigationSettings.voiceVolume)
+        let muteChanged = notification.userInfo?[muteKey] != nil
```

Comment:
If this notification could be subscribed to in more than one place, consider refactoring these into computed properties in an extension.